pkg/destroy/openstack/openstack.go

```diff
@@ -998,3 +1009,63 @@ func deleteImages(opts *clientconfig.ClientOpts, filter Filter, logger logrus.Fi
 	}
 	return true, nil
 }
+
+func untagRunner(opts *clientconfig.ClientOpts, networkTag string, logger logrus.FieldLogger) error {
+	backoffSettings := wait.Backoff{
+		Duration: time.Second * 10,
+		Steps:    25,
+	}
+
+	err := wait.ExponentialBackoff(backoffSettings, func() (bool, error) {
+		return untagNetwork(opts, networkTag, logger)
+	})
+	if err != nil {
+		return errors.Errorf("Unrecoverable error/timed out: %v", err)
+	}
+
+	return nil
+}
+
+// untagNetworks removes a tag from the network
+func untagNetwork(opts *clientconfig.ClientOpts, networkTag string, logger logrus.FieldLogger) (bool, error) {
+	logger.Debug("Untagging openstack networks")
```

Comment:
can you add the infra id to the log message?

Reply:
done